### PR TITLE
Backend/purchases/transaction

### PR DESCRIPTION
**Pull request by ghost**
+ Took a bit longer than expected, added the purchase pass option, which allows users to select the pass they want, check out documentation updates in following commits.(Still need to implement a listener)
+ Added a few properties to the user node in preparation for ratings  (totalStars and numberOfRatings)
+ Added the acceptedSale property to pass nodes to manage the state of the pass for sale (such as initial state , pending and accepted)
+ Updated activeUsers endpoint to return the users average rating  …
+ Updated tests to reflect activeUsers endpoint.
+ Updated readme to reflect changed endpoints

### CODE REVIEW

File: db_operations/db.js [hunk 1]
```diff
@@ -431,56 +442,155 @@ const registerPass = (email, api_key, lotLocation, gpsLocation, price, notes, ca
...
+            var passObject = new Object();
+            passObject.avgRating = parseFloat(record.get('totalStars')) / parseFloat(record.get('numberOfRatings'));
+            passObject.gpsLocation = record.get('gpsLocation');
+            passObject.notes = record.get('notes');
+            passObject.forSale = record.get('forSale');
+            passObject.price = record.get('price');
+            passObject.lotLocation = record.get('lotLocation');
+            passObject.passId = record.get('passId');
+            passObject.ownerEmail = record.get('ownerEmail');
+            passObject.ownerCount = record.get('ownerCount');
```
Comment (RGrab) [inline db_operations/db.js]: The front end doesn't have a need for 'ownerCount' value at this moment. would you be able to comment this out forthe time being.

**Review by RGrab — CHANGES_REQUESTED** [review]
The code looks solid from what I can see just change that one thing. good job! :1st_place_medal:

File: db_operations/db.js (hunk 1 above)
Comment (Lahpman7) [inline db_operations/db.js]: No problem. Handling it now.

**Comment by Lahpman7** [comment]
The property being returned has been removed.

**Review by RGrab — APPROVED** [review]

**Pull request merged**